aea/cli/eject.py

```diff
@@ -0,0 +1,124 @@
+from aea.cli.utils.package_utils import (
+    copy_package_directory,
+    get_package_eject_path,
+    get_package_vendor_path,
+    is_item_present,
+)
+from aea.configurations.base import DEFAULT_VERSION, PublicId
+
+
+@click.group()
+@click.pass_context
+@check_aea_project
+def eject(click_context: click.core.Context):
+    """Eject an installed item."""
+
+
+@eject.command()
+@click.argument("public_id", type=PublicIdParameter(), required=True)
+@pass_ctx
+def connection(ctx: Context, public_id: PublicId):
+    """Eject an installed connection."""
+    _eject_item(ctx, "connection", public_id)
+
+
+@eject.command()
+@click.argument("public_id", type=PublicIdParameter(), required=True)
+@pass_ctx
+def contract(ctx: Context, public_id: PublicId):
+    """Eject an installed contract."""
+    _eject_item(ctx, "contract", public_id)
+
+
+@eject.command()
+@click.argument("public_id", type=PublicIdParameter(), required=True)
+@pass_ctx
+def protocol(ctx: Context, public_id: PublicId):
+    """Eject an installed protocol."""
+    _eject_item(ctx, "protocol", public_id)
+
+
+@eject.command()
+@click.argument("public_id", type=PublicIdParameter(), required=True)
+@pass_ctx
+def skill(ctx: Context, public_id: PublicId):
+    """Eject an installed skill."""
+    _eject_item(ctx, "skill", public_id)
+
+
+@clean_after
+def _eject_item(ctx: Context, item_type: str, public_id: PublicId):
+    """
+    Eject item from installed (vendor) to custom folder.
+
+    :param ctx: context object.
+    :param item_type: item type.
+    :param public_id: item public ID.
+
+    :return: None
+    :raises: ClickException if item is absent at source path or present at destenation path.
+    """
+    item_type_plural = item_type + "s"
+    supported_items = getattr(ctx.agent_config, item_type_plural)
+    if (
+        not is_item_present(ctx, item_type, public_id)
+        or public_id not in supported_items
+    ):
+        raise click.ClickException(
+            "{} {} not found in agent items.".format(item_type.title(), public_id)
+        )
+    src = get_package_vendor_path(ctx, item_type, public_id)
+    dst = get_package_eject_path(ctx, item_type, public_id)
+    if is_item_present(ctx, item_type, public_id, ejected=True):
+        raise click.ClickException(
+            "{} {} is already in agent ejected items.".format(
+                item_type.title(), public_id
+            )
+        )
+
+    ctx.clean_paths.append(dst)
+    copy_package_directory(Path(src), dst)
+
+    try_to_load_agent_config(ctx)
+    new_public_id = PublicId(
+        author=ctx.agent_config.author, name=public_id.name, version=DEFAULT_VERSION
+    )
+    update_item_config(
+        item_type, Path(dst), author=new_public_id.author, version=new_public_id.version
+    )
+    supported_items.add(new_public_id)
+    supported_items.remove(public_id)
+    update_item_config("agent", Path(ctx.cwd), **{item_type_plural: supported_items})
```

Comment:
Looks ok but we definitely have implemented similar functionality elsewhere...

Reply:
See above.